### DISCUSSION TITLE
Breaking R2R (Crossgen2) per-app version bubble promise

### DISCUSSION THREAD

**Discussion by vuplea** [dotnet/runtime#70166] [Q&A] [answered]
Using cross-assemblies version bubbles requires the user to not mess with the assembly versions before they will run.

If one were to do that (let's say, replace a assembly in the bubble with a newer assembly version), what is the expected behavior?
Is it a fail-fast, fallback to IL, or something unpredictable?

**Comment by jkotas** [comment, marked as answer]
The expected behavior is fail-fast.
